autogen/telemetry.py

```diff
@@ -0,0 +1,348 @@
+from __future__ import annotations
+
+import datetime
+import inspect
+import json
+import logging
+import sqlite3
+import sys
+import uuid
+
+from typing import TYPE_CHECKING, Any, Dict, List, Union, Tuple
+
+if TYPE_CHECKING:
+    from autogen import ConversableAgent, OpenAIWrapper
+
+try:
+    import openai
+except ImportError:
+    ERROR = ImportError("Please install openai>=1 and diskcache to use autogen.OpenAIWrapper.")
+    OpenAI = object
+    AzureOpenAI = object
+else:
+    from openai import OpenAI, AzureOpenAI
+    from openai.types.chat import ChatCompletion
+
+    ERROR = None
+
+# this is a pointer to the module object instance itself
+this = sys.modules[__name__]
+this._session_id = None
+this._con = None
+this._cur = None
+logger = logging.getLogger(__name__)
+
+
+def start_logging(dbname: str = "telemetry.db") -> str:
```

Comment:
Very soon we will get request or PR contribution to support other people's favorite databases.  

Consider making argument list "forward compatible" so it is easy to add new database support while not changing the interface method, as backward compatibility is important.

Reply:
E.g. use connection string